### ISSUE TITLE
[FEATURE] Add Installer Option to disable TRIM

### ISSUE THREAD

**Issue by PeterBurner** [osy/HaC-Mini#998]
**Description**
After upgrading to macOS Monterey I was experiencing very long boot times (4min+). After some digging I found that macOS Monterey has problems with the TRIM functionality of my Samsung NVMe SSD (Samsung 970 EVO Plus 500GB).
I found some threads in different forums that suggested to disable the TRIM functionality. For me this helped bring down the boot time by a lot.

**Suggestion**
 - Update OpenCore to >=  0.7.9
 - Add installer option to set `Kernel > Quirks > SetApfsTrimTimeout` to `0`

**Environment:**
 - Running OSX version: 12.3.1 (21E258)
 - HaC Mini installer version: 3.1.5

**Additional context**
 - https://github.com/dortania/bugtracker/issues/192
 - https://www.hackintosh-forum.de/forum/thread/55361-opencore-slow-boot/
 
 From OpenCore 0.7.9 release notes:
> Added new method to disable trim when SetApfsTrimTimeout is set to zero

**Comment by StSiem** [comment]
I also have the problem of a significantly increased boot time since upgrading to macOS Monterey. And I also have a Samsung 970 EVO.

Turning off TRIM to get normal boot times sounds great, but aren't there (significant) downsides to doing so? A quick web search suggested, that it may decrease the life time of your SSD.

**Comment by PeterBurner** [comment]
@StSiem I have the same concern. However the question is, does TRIM work in the first place? As far as I understand it the TRIM command does not work properly and runs into a timeout. After that delay the boot process continues. So there would be no difference in SSD lifetime if we disable TRIM.
I am no expert on this. @osy86 feel free to correct me.
Either way its probably best to replace the SSD with a supported model if one is able to do so (I am currently not).

**Comment by StSiem** [comment]
It would be interesting to know if users with other SSD models and TRIM enabled are also affected by the long boot time 🤔

**Comment by PeterBurner** [comment]
If someone is looking for a replacement SSD, the [issue](https://github.com/dortania/bugtracker/issues/192) I linked above contains a list of compatible models.

**Comment by StSiem** [comment]
Ah perfect. Thank you for the link.

**Comment by Medizinmann-MD** [comment]
> **Description** After upgrading to macOS Monterey I was experiencing very long boot times (4min+). After some digging I found that macOS Monterey has problems with the TRIM functionality of my Samsung NVMe SSD (Samsung 970 EVO Plus 500GB). I found some threads in different forums that suggested to disable the TRIM functionality. For me this helped bring down the boot time by a lot.
> 

Does your drive already use the latest firmware?

**Comment by Medizinmann-MD** [comment]
Well I did some research - and it is a mess....

First thing – it seems Apple drivers work best with Phison Controllers (i.e. used in WD Black drives or Crucial) and doesn’t work very well with the Controllers used by Samsung (Phoenixs) – Samsung made a Firmware Update to the 970 EVO Plus in 2019 that gave support for MacOS wich broke with Apples NVME implementation in Monterey…
The next bad thing is that Samsung changed the Controllers to from Phoenix to Elpin in late 2021 and we most probably won’t see any more Firmware Updates for the older drives…

And some people report success with newer Elpin drives with latest Firmware – but I am not too sure about it…

To add more bad while WD Black drives seem to work fine they are actually slower than the Samsung drives – but of course still much faster when working with working TRIM support with Monterey than the broken Driver support with Samsung drives…

tl;dr
Deactivating TIRM is only a stopgap solutions.
Right now the only “real” solution is to use something like WD Black 750 or 550 instead - just odered some...

**Comment by danieldeicide** [comment]
> Samsung made a Firmware Update to the 970 EVO Plus in 2019 that gave support for MacOS wich broke with Apples NVME implementation in Monterey

@Medizinmann-MD 
This means that an NVMe drive from Samsung with updated firmware (let's say Samsung 980 PRO) should work on Big Sur?
Thank you!

**Comment by Medizinmann-MD** [comment]
> > Samsung made a Firmware Update to the 970 EVO Plus in 2019 that gave support for MacOS wich broke with Apples NVME implementation in Monterey
> 
> @Medizinmann-MD This means that an NVMe drive from Samsung with updated firmware (let's say Samsung 980 PRO) should work on Big Sur? Thank you!

Well - AFAIK Big Sur should work fine with 970 EVO and 970 EVO Plus (see https://dortania.github.io/Anti-Hackintosh-Buyers-Guide/Storage.html). 

With the 980 Pro even some people report success with Monterey - but Ia m not so sure about that - I don't have very reliable sources here....

**Comment by wonsage** [comment]
I have the same problem, but the Kingston nvme SSD.

**Comment by osy** [comment]
@PeterBurner please test https://github.com/osy/HaC-Mini/suites/9532886939/artifacts/453073226

**Comment by BjoernSaja** [comment]
Thanks @osy!
I installed your test version on top of the latest HaCMini release. I selected `SetNoTrim` in setup wizard.
I can confirm that my Hackintosh still boots and that the boot time is greatly reduced like when setting `SetApfsTrimTimeout` manually.
Many thanks for your effort!

**Comment by VavyTech** [comment]
qual o comando para eu desativar o trim com opecore?
